api-compact.yaml

```diff
@@ -199,3 +274,45 @@ endpoints:
           x-amazon-apigateway-integration:
             <<: *apigateway_integration_with_id_and_querystring_parameters
             uri: "https://${stageVariables.enterprise_host}/enterprise/api/v1/catalogs/{id}/courses/"
+
+    # /v1/enterprise_catalogs/
+    newEnterpriseCatalogs:
+        get:
+          produces: *produces
+          parameters:
+            - *auth_header
+            - *limit_qs_parameter
+            - *offset_qs_parameter
+            - *page_qs_parameter
+            - *page_size_qs_parameter
+          operationId: "new_get_enterprise_catalogs"
+          responses: *responses
+          x-amazon-apigateway-integration:
+            <<: *apigateway_integration_uuid
+            uri: "https://${stageVariables.enterprise_host}/enterprise/api/v1/enterprise_catalogs/"
+
+    # /v1/enterprise_catalogs/{uuid}/
+    enterpriseCatalogByUuid:
+        get:
+          produces: *produces
+          parameters:
+            - *auth_header
+            - *uuid_parameter
+          operationId: "get_enterprise_catalog_by_uuid"
+          responses: *responses
+          x-amazon-apigateway-integration:
+            <<: *apigateway_integration_with_uuid_parameter
+            uri: "https://${stageVariables.enterprise_host}/enterprise/api/v1/enterprise_catalogs/{uuid}/"
+
+    # /v1/programs/{uuid}/
+    programByUuid:
+        get:
+          produces: *produces
+          parameters:
+            - *auth_header
+            - *uuid_parameter
+          operationId: "get_program_by_uuid"
+          responses: *responses
+          x-amazon-apigateway-integration:
+            <<: *apigateway_integration_with_uuid_parameter
+            uri: "https://${stageVariables.enterprise_host}/enterprise/api/v1/programs/{uuid}/"
```

Comment:
@itsjeyd @brittneyexline Then we're really in trouble because I'm in the same boat, too, even though I wrote this. Let's just say I kinda winged it hoping someone knowledgeable in review would point out my possibly incorrect assumptions :)

@saleem-latif Can you review these api-compact.yml changes and let me know if they make sense given the endpoints we're trying to put into the gateway?